source/app/blueprints/dim_tasks/dim_tasks.py

```diff
@@ -0,0 +1,314 @@
+    if not hook_ui_name:
+        return response_error('Missing hook_ui_name')
+
+    targets = js_data.get('targets')
+    if not targets:
+        return response_error('Missing targets')
+
+    data_type = js_data.get('type')
+    if not data_type:
+        return response_error('Missing data type')
+
+    index = 0
+    obj_targets = []
+    for target in js_data.get('targets'):
+        obj = None
+        if data_type == 'ioc':
+            obj = Ioc.query.filter(Ioc.ioc_id == target).first()
+
+        elif data_type == "case":
+            obj = Cases.query.filter(Cases.case_id == caseid).first()
+
+        elif data_type == "asset":
+            obj = CaseAssets.query.filter(
+                    CaseAssets.asset_id == target,
+                    CaseAssets.case_id == caseid
+            ).first()
+
+        elif data_type == "note":
+            obj = Notes.query.filter(
+                    Notes.note_id == target,
+                    Notes.note_case_id == caseid
+            ).first()
+
+        elif data_type == "event":
+            obj = CasesEvent.query.filter(
+                    CasesEvent.event_id == target,
+                    CasesEvent.case_id == caseid
+            ).first()
+
+        elif data_type == "task":
+            obj = CaseTasks.query.filter(
+                    CaseTasks.id == target,
+                    CaseTasks.task_case_id == caseid
+            ).first()
+
+        elif data_type == "evidence":
+            obj = CaseReceivedFile.query.filter(
+                    CaseReceivedFile.id == target,
+                    CaseReceivedFile.case_id == caseid
+            ).first()
+
+        elif data_type == "global_task":
+            obj = GlobalTasks.query.filter(
+                    GlobalTasks.id == target
+            ).first()
+
+        else:
+            logs.append(f'Data type {data_type} not supported')
+            continue
+
+        if not obj:
+            logs.append(f'Object ID {target} not found')
+            continue
+        obj_targets.append(obj)
+
+        # Call to queue task
+        index += 1
+
+    if len(obj_targets) > 0:
+        call_modules_hook(hook_name=hook_name, hook_ui_name=hook_ui_name, data=obj_targets, caseid=caseid)
+
+    if len(logs) > 0:
+        return response_error(f"Errors encountered during processing of data. Queued task with {index} objects",
+                              data=logs)
+
+    return response_success(f'Queued task with {index} objects')
+
+
+@dim_tasks_blueprint.route('/dim/tasks/list', methods=['GET'])
+@api_login_required
+def list_dim_tasks(caseid):
+    tasks = CeleryTaskMeta.query.with_entities(
+        CeleryTaskMeta.task_id,
+        CeleryTaskMeta.date_done
+    ).order_by(desc(CeleryTaskMeta.date_done)).limit(200).all()
+
+    data = []
+
+    for row in tasks:
+        task = app.celery.AsyncResult(row.task_id)
+
+        task_name = task.name if task.name else "No engine. Unrecoverable shadow failure"
```

Comment:
Issue linked to loading of legacy tasks because of change on IITaskStatus:
```
AttributeError: Can't get attribute 'IITaskStatus' on <module 'iris_interface.IrisInterfaceStatus' from '/opt/venv/lib/python3.9/site-packages/iris_interface/IrisInterfaceStatus.py'>
```